app/models/room.rb

```diff
@@ -16,7 +16,13 @@ class Room < ApplicationRecord
   validates :meeting_id, presence: true, uniqueness: true
 
   before_validation :set_friendly_id, :set_meeting_id, on: :create
+
   after_create :create_meeting_options
+  after_create :generate_viewer_access_code, if: :auto_generate_viewer_access_code?
+  after_create :generate_moderator_access_code, if: :auto_generate_moderator_access_code?
```

Comment:
Why wouldn't we just move this logic into the `create_meeting_options`?

Reply:
`create_meeting_options` will launch only after room creation.
The access codes auto generation case isn't limited to creation but also to the first fetching opertation (find).
Besides the access codes generation on room creation requires having the room meeting options in place therefore splitting that logic into a separate callback will reduce the complexity of `create_meeting_options` and improve readability.

More explanation on this is given above.